src/TriangulationFactory.cpp

```diff
@@ -1004,6 +1004,11 @@ Triangulation* TriangulationFactory::cylinder(Arena* arena, const Geometry* geo,
 
 Triangulation* TriangulationFactory::sphereBasedShape(Arena* arena, const Geometry* geo, float radius, float arc, float shift_z, float scale_z, float scale)
 {
+
+  if (scale_z == INFINITY) {
```

Comment:
Please use `if(!std::isfinite(scale_z))`, then you catch both infinity and nan, both positive and negative.

Reply:
thanks, fixed